geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -6847,6 +6847,10 @@ public void txApplyDestroy(Object key, TransactionId rmtOrigin, TXRmtEvent event
     }
   }
 
+  boolean isInTokenModeNeeded(boolean needTokensForGII, boolean needRIDestroyToken) {
+    return (needTokensForGII || needRIDestroyToken) && !getConcurrencyChecksEnabled();
```

Comment:
Small change, but can you swap the check around? First evaluate the ConcurrencyCheck. As this check has a greater chance to short circuit the conditional sooner.